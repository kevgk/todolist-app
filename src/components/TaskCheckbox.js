import React from 'react';

export default function TaskCheckbox({ clickHandler, isChecked }) {
  return (
    <button className='checkbox' onClick={() => clickHandler()}>
      <span role="img" aria-label="check task" style={{ opacity: isChecked ? 1 : 0.1}}>âœ”</span>
    </button>
  );
}