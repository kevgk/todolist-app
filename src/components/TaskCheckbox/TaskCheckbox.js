import React from 'react';
import styles from './TaskCheckbox.module.css';

export default function TaskCheckbox({ checkHandler, isChecked }) {
  return (
    <button className={styles.checkbox} onClick={() => checkHandler()}>
      <span role="img" aria-label="check task" style={{ opacity: isChecked ? 1 : 0.1}}>âœ”</span>
    </button>
  );
}