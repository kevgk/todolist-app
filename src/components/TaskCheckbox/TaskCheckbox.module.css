.checkbox {
  width: 20px;
  height: 20px;
  border: none;
  border-radius: 5px;
  background-color: #7047f5;
  cursor: pointer;
  display: flex;
  justify-content: center;
}

.checkbox > span {
  transition: opacity 0.3s;
  color: #fff;
}