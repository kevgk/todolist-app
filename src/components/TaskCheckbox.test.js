import React from 'react';
import { mount } from 'enzyme';

import TaskCheckbox from './TaskCheckbox';

it('renders correctly (unchecked)', () => {
	const wrapper = mount(
		<TaskCheckbox clickHandler={() => {}} isChecked={false} />
	);

	expect(wrapper).toMatchSnapshot();
});

it('renders correctly (checked)', () => {
	const wrapper = mount(
		<TaskCheckbox clickHandler={() => {}} isChecked={true} />
	);

	expect(wrapper).toMatchSnapshot();
});
