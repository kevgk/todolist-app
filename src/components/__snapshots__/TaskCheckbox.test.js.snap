// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly (checked) 1`] = `
<button
  className="checkbox"
  onClick={[Function]}
>
  <span
    aria-label="check task"
    role="img"
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    ✔
  </span>
</button>
`;

exports[`renders correctly (unchecked) 1`] = `
<button
  className="checkbox"
  onClick={[Function]}
>
  <span
    aria-label="check task"
    role="img"
    style={
      Object {
        "opacity": 0.1,
      }
    }
  >
    ✔
  </span>
</button>
`;
